### PR TITLE
add CanardTxTransfer object for sending payloads

### PR DESCRIPTION
**Pull request by bugobliterator**
This should make it easier to add more options while sending payloads.

### CODE REVIEW

File: canard.c [hunk 1]
```diff
@@ -150,11 +150,35 @@ int16_t canardBroadcast(CanardInstance* ins,
...
+#endif
+#if CANARD_ENABLE_CANFD
+        .canfd = canfd,
+#endif
+    };
+
+    return canardBroadcastObj(ins, &transfer_object);
+}
+
+int16_t canardBroadcastObj(CanardInstance* ins, CanardTxTransfer* transfer_object)
```
Comment (tridge) [inline canard.c:175]: should have a comment on why using this API

File: canard.h [hunk 2]
```diff
@@ -228,6 +228,29 @@ typedef struct CanardInstance CanardInstance;
 typedef struct CanardRxTransfer CanardRxTransfer;
 typedef struct CanardRxState CanardRxState;
 typedef struct CanardTxQueueItem CanardTxQueueItem;
+
+typedef struct {
```
Comment (tridge) [inline canard.h:232]: this should explain what the structure is for

File: canard.h [hunk 3]
```diff
@@ -478,6 +501,9 @@ int16_t canardBroadcast(CanardInstance* ins,            ///< Library instance
 #endif
                         );
 
+int16_t canardBroadcastObj(CanardInstance* ins,            ///< Library instance
```
Comment (tridge) [inline canard.h:504]: should explain why to use this instead of canardBroadcast

File: canard.h [hunk 4]
```diff
@@ -516,6 +542,10 @@ int16_t canardRequestOrRespond(CanardInstance* ins,             ///< Library ins
 #endif
                             );
 
+int16_t canardRequestOrRespondObj(CanardInstance* ins,             ///< Library instance
```
Comment (tridge) [inline canard.h:545]: needs explanation

**Review by tridge — CHANGES_REQUESTED** [review]

File: canard.c (hunk 1 above)
Comment (bugobliterator) [reply canard.c:182]: The details are in .h file, all API details are in header files not source

**Review by tridge — APPROVED** [review]

**Pull request merged**